client/src/resources/QConfig.js

```diff
@@ -1,9 +1,21 @@
+import { inject } from "aurelia-framework";
 import qEnv from "resources/qEnv.js";
+import User from "resources/User.js";
+import { HttpClient } from "aurelia-fetch-client";
 
+@inject(User, HttpClient)
 export default class QConfig {
-  constructor() {
-    this.configLoaded = qEnv.QServerBaseUrl.then(QServerBaseUrl => {
-      return fetch(`${QServerBaseUrl}/editor/config`);
+  constructor(user, httpClient) {
+    this.httpClient = httpClient;
+    this.configLoaded = new Promise(resolve => {
+      this.resolveConfigLoaded = resolve;
+    });
+    user.loaded.then(() => {
+      return qEnv.QServerBaseUrl.then(QServerBaseUrl => {
```

Comment:
A quick question regarding the init lifecycle in aurelia. Isn't it possible to hook into a before load method to fetch such values? If we have that, we wouldn't need to use `await` everywhere to load the confiuguration.

Reply:
Not sure if I correctly understand your intention here. You suggest to sync load all the config before aurelia starts up?

Reply:
Aren't all the configs of `qEnv` loaded at once? So yes, simplifying that and then supporting sync access is easier. I've seen a few workarounds in the codebase just because of the async behavior.

Reply:
Not necessary beore aurelia starts up, but before the first controller gets initialized. Maybe when injecting the variable?

Reply:
Ah, yes that could be refactored. I would argue to not do that now though since we already diverged from upstream and should imo not go too far down that road if not really necessary. Do you agree?

Reply:
yes, of course. I was just wondering about the workarounds and how it was able to live that long 😁

Reply:
well, I call it hashtag-fokus sometimes. When there are so many things to be improved, one has to make sure to pick to most important ones to work on...